### DISCUSSION TITLE
Equivalent of dnsmasq domain-needed

### DISCUSSION THREAD

**Discussion by timkgh** [AdguardTeam/AdGuardHome#3587] [Q&A] [answered]
Is there an equivalent of dnsmasq `domain-needed` in AGH or does it expect a FQDN by default?

I do not want queries without a domain to end up at the upstream servers.

**Comment by ainar-g** [comment, marked as answer]
Hello.  We support the `[//]` syntax of `dnsmasq` for unqualified domain names, so you can just put something like

```none
[//]127.0.0.1
```

into the upstreams inputs, and all requests like `nslookup foo` should fail.

**Comment by timkgh** [comment]
Great. I actually want to send it to the local dnsmasq which handles it.

Followup question, is it possible to default a domain in AGH if not set? E.g. a query for `foo` to be handled as `foo.mydomain`? Just like `/etc/resolv.conf` supports `domain` or `search` directives.

**Reply by ainar-g** [reply]
As far as I know, currently this is not possible.  You can add `$dnsrewrite` custom rules to handle some of these manually though.
